plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQuerySplitManager.java

```diff
@@ -81,11 +80,10 @@ public BigQuerySplitManager(
     {
         this.bigQueryClientFactory = requireNonNull(bigQueryClientFactory, "bigQueryClientFactory cannot be null");
         this.bigQueryReadClientFactory = requireNonNull(bigQueryReadClientFactory, "bigQueryReadClientFactory cannot be null");
-        this.parallelism = config.getParallelism();
+        this.actualParallelism = config.getParallelism().orElseGet(() -> requireNonNull(nodeManager, "nodeManager cannot be null").getRequiredWorkerNodes().size());
```

Comment:
If we want to do it this way we should use memoized supplier

Reply:
Let's just revert the commit.